### ISSUE TITLE
useREST API: SVM Name mandatory in the backend

### ISSUE THREAD

**Issue by YvosOnTheHub** [NetApp/trident#612]
**Describe the bug**
Until now, the _svm_ parameter was optional as stated in documentation
https://netapp-trident.readthedocs.io/en/stable-v21.07/kubernetes/operations/tasks/backends/ontap/ontap-nas/configuration.html?highlight=Derived#ontap-nas-backend-configuration
=> Derived if an SVM managementLIF is specified

With Trident 21.07 & useREST set, Trident does not retrieve the SVM name anymore

**Environment**
Lab on Demand Kubernetes v4.0

- Trident version: 21.07.0
- Trident installation flags used: standard with Helm
- Container runtime: Docker 18.09.1
- Kubernetes version: 1.18.6
- Kubernetes orchestrator: Kubernetes
- OS: RHEL 7.5
- NetApp backend types: ONTAP 9.8

**To Reproduce**
Create a backend without the parameter _svm_ set & with the parameter useREST:true

**Expected behavior**
Either modify documentation to make this parameter mandatory when using REST, or fix Trident to retrieve the SVM name.

**Additional context**
Difference between a ZAPI backend & a REST backend (both **without** the _svm_ parameter set in the yaml backend file)
```
[root@rhel3 K8S_objects]# trident get backend nas-zapi -o json | grep -i svm
        "svm": "nfs_svm",

[root@rhel3 K8S_objects]# trident get backend nas-rest -o json | grep -i svm
        "svm": "",
```

Logs from Trident are pretty explicit:

POST /api/storage/volumes?return_records=true HTTP/1.1
Host: 192.168.0.135
User-Agent: Go-http-client/1.1
Content-Length: 372
Accept: application/hal+json
Accept: application/json
Authorization: Basic dnNhZG1pbjpOZXRhcHAxIQ==
Content-Type: application/hal+json
Accept-Encoding: gzip

{"aggregates":[{"name":"aggr1"}],"comment":"","guarantee":{"type":"none"},"name":"rest_pvc_ca445898_c421_44be_9a19_da63dc61c17a","nas":{"export_policy":{"name":"default"},"security_style":"unix","unix_permissions":777},"size":5651272757,"snapshot_policy":{"name":"default"},"space":{"snapshot":{"reserve_percent":5}},"state":"online","svm":{},"tiering":{"policy":"none"}}

HTTP/1.1 400 Bad Request
Connection: close
Content-Length: 114
Cache-Control: no-cache,no-store,must-revalidate
Content-Type: application/hal+json
Date: Tue, 03 Aug 2021 07:11:46 GMT
Server: libzapid-httpd
X-Content-Type-Options: nosniff

{
  "error": {
    "message": "Either \"svm.uuid\" or \"svm.name\" must be provided.",
    "code": "2621707"
  }
}
time="2021-08-03T07:11:47Z" level=error msg="ONTAP-NAS pool aggr1/aggr1; error creating volume rest_pvc_ca445898_c421_44be_9a19_da63dc61c17a: error creating volume: [POST /storage/volumes][400] volume_create default  &{Error:0xc00056f040}" requestID=64af7cfc-3a46-42a8-90ce-47d63604d0e9 requestSource=CSI
time="2021-08-03T07:11:47Z" level=warning msg="Failed to create the volume on this pool." backend=nas-rest backendUUID=6a72cf66-bc01-4626-971b-0ec3e4729fdb error="backend cannot satisfy create request for volume rest_pvc_ca445898_c421_44be_9a19_da63dc61c17a: (ONTAP-NAS pool aggr1/aggr1; error creating volume rest_pvc_ca445898_c421_44be_9a19_da63dc61c17a: error creating volume: [POST /storage/volumes][400] volume_create default  &{Error:0xc00056f040})" pool=aggr1 requestID=64af7cfc-3a46-42a8-90ce-47d63604d0e9 requestSource=CSI volume=pvc-ca445898-c421-44be-9a19-da63dc61c17a
time="2021-08-03T07:11:47Z" level=debug msg="Volume event." eventType=Normal message="encountered error(s) in creating the volume: [Failed to create volume pvc-ca445898-c421-44be-9a19-da63dc61c17a on storage pool aggr1 from backend nas-rest: backend cannot satisfy create request for volume rest_pvc_ca445898_c421_44be_9a19_da63dc61c17a: (ONTAP-NAS pool aggr1/aggr1; error creating volume rest_pvc_ca445898_c421_44be_9a19_da63dc61c17a: error creating volume: [POST /storage/volumes][400] volume_create default  &{Error:0xc00056f040})]" name=pvc-ca445898-c421-44be-9a19-da63dc61c17a reason=ProvisioningFailed requestID=64af7cfc-3a46-42a8-90ce-47d63604d0e9 requestSource=CSI
time="2021-08-03T07:11:47Z" level=debug msg="Found cached PVC by UID." name=pvc-rest namespace=default requestID=64af7cfc-3a46-42a8-90ce-47d63604d0e9 requestSource=CSI uid=ca445898-c421-44be-9a19-da63dc61c17a
time="2021-08-03T07:11:47Z" level=debug msg="<<<< CreateVolume" Method=CreateVolume Type=CSI_Controller name=pvc-ca445898-c421-44be-9a19-da63dc61c17a requestID=64af7cfc-3a46-42a8-90ce-47d63604d0e9 requestSource=CSI
time="2021-08-03T07:11:47Z" level=error msg="GRPC error: rpc error: code = Unknown desc = encountered error(s) in creating the volume: [Failed to create volume pvc-ca445898-c421-44be-9a19-da63dc61c17a on storage pool aggr1 from backend nas-rest: backend cannot satisfy create request for volume rest_pvc_ca445898_c421_44be_9a19_da63dc61c17a: (ONTAP-NAS pool aggr1/aggr1; error creating volume rest_pvc_ca445898_c421_44be_9a19_da63dc61c17a: error creating volume: [POST /storage/volumes][400] volume_create default  &{Error:0xc00056f040})]" requestID=64af7cfc-3a46-42a8-90ce-47d63604d0e9 requestSource=CSI

**Comment by gnarl** [comment]
This issue is fixed with [commit 800db19](https://github.com/NetApp/trident/commit/800db19fdb08fd953ee35ec2895f0bbb324cd5c9) and is included in the Trident 21.10.1 release.

**Comment by yasuoohno** [comment]
Hi, I have same issue with Trident 21.10.1 and 22.01.0.
I can create backend but a volume cannot be created.
Should I report this as a separate issue? 

```
{
  "error": {
    "message": "Either \"svm.uuid\" or \"svm.name\" must be provided.",
    "code": "2621707"
  }
}
```

These logs are from 22.01.0.

> time="2022-02-02T00:48:01Z" level=debug msg="POST /api/storage/volumes HTTP/1.1\r\nHost: 192.168.3.155\r\nUser-Agent: Go-http-client/1.1\r\nContent-Length: 427\r\nAccept: application/hal+json\r\nAccept: application/json\r\nAuthorization: Basic dnNhZG1pbjpuZXRhcHAxMjM=\r\nContent-Type: application/hal+json\r\nAccept-Encoding: gzip\r\n\r\n{\"aggregates\":[{\"name\":\"aggr1\"}],\"comment\":\"\",\"guarantee\":{\"type\":\"none\"},\"name\":\"trident_pvc_f4933db8_31f3_4018_a60f_71d1f28d28dd\",\"nas\":{\"export_policy\":{\"name\":\"trident-cfa8bcdb-9e55-43b2-b796-4ea9ae3a6bff\"},\"security_style\":\"unix\",\"unix_permissions\":777},\"size\":1073741824,\"snapshot_policy\":{\"name\":\"none\"},\"space\":{\"snapshot\":{\"reserve_percent\":0}},\"state\":\"online\",\"style\":\"flexvol\",\"svm\":{},\"tiering\":{\"policy\":\"none\"}}\n\n"
> time="2022-02-02T00:48:02Z" level=debug msg="HTTP/1.1 400 Bad Request\r\nConnection: close\r\nContent-Length: 114\r\nCache-Control: no-cache,no-store,must-revalidate\r\nContent-Security-Policy: default-src 'self'; script-src 'self' 'unsafe-inline'; style-src 'self' 'unsafe-inline'; img-src 'self' data:; frame-ancestors: 'self'\r\nContent-Type: application/hal+json\r\nDate: Wed, 02 Feb 2022 00:48:02 GMT\r\nServer: libzapid-httpd\r\nX-Content-Type-Options: nosniff\r\n\r\n{\n  \"error\": {\n    \"message\": \"Either \\\"svm.uuid\\\" or \\\"svm.name\\\" must be provided.\",\n    \"code\": \"2621707\"\n  }\n}\n"
> time="2022-02-02T00:48:02Z" level=error msg="ONTAP-NAS pool aggr1/aggr1; error creating volume trident_pvc_f4933db8_31f3_4018_a60f_71d1f28d28dd: error creating volume: [POST /storage/volumes][400] volume_create default  &{Error:0xc00064a140}" requestID=ebce7f9d-171a-4c0e-bc06-ecaf64178289 requestSource=CSI
> time="2022-02-02T00:48:02Z" level=warning msg="Failed to create the volume on this pool." backend=ontap-nas backendUUID=cfa8bcdb-9e55-43b2-b796-4ea9ae3a6bff error="backend cannot satisfy create request for volume trident_pvc_f4933db8_31f3_4018_a60f_71d1f28d28dd: (ONTAP-NAS pool aggr1/aggr1; error creating volume trident_pvc_f4933db8_31f3_4018_a60f_71d1f28d28dd: error creating volume: [POST /storage/volumes][400] volume_create default  &{Error:0xc00064a140})" pool=aggr1 requestID=ebce7f9d-171a-4c0e-bc06-ecaf64178289 requestSource=CSI volume=pvc-f4933db8-31f3-4018-a60f-71d1f28d28dd

**Comment by gnarl** [comment]
@yasuoohno, thanks for bringing this to our attention. I'm reopening this issue since the original issue wasn't fixed for all code paths.

**Comment by gnarl** [comment]
This issue was fixed with [commit 800db19](https://github.com/NetApp-openstack-dev/trident/commit/800db19fdb08fd953ee35ec2895f0bbb324cd5c9) and is available in the Trident v22.01 release.
